src/polycap.c

```diff
@@ -864,6 +918,108 @@ struct calcstruct *init_calcstruct(unsigned long int seed, struct cap_profile *p
+			pc_x[0] = 0.;
+			pc_y[0] = rad_ext[0];
+			pc_x[3] = length;
+			pc_y[3] = rad_ext[1];
+			if(focal_dist[0] <= length) pc_x[1] = focal_dist[0]/10.;
+				else pc_x[1] = length/10.; 
+			pc_y[1] = (rad_ext[0]-0.)/(0.-(-1.*focal_dist[0])) * (pc_x[1] - 0.) + rad_ext[0]; //extrapolate line between focus point and PC entrance
+			if(focal_dist[1] <= length) pc_x[2] = length-focal_dist[0]/10.;
+				else pc_x[2] = length-length/10.; 
+			pc_y[2] = (rad_ext[1]-0.)/(length-(length+focal_dist[1])) * (pc_x[2] - length) + rad_ext[1]; //extrapolate line between focus point and PC exit
+			polynomialfit(4, 3, pc_x, pc_y, coeff);
+
+			//calculate shape coordinates
+			for(i=0;i<profile->nmax;i++){
+				profile->arr[i].zarr = length/profile->nmax*i; //z coordinates, from 0 to length
+				profile->arr[i].profil = (rad_int[1]-rad_int[0])/length*profile->arr[i].zarr + rad_int[0]; //single capillary shape always conical
+				profile->arr[i].sx = 0; //set sx and sy to 0 as they are overwritten in start() anyway. 
+				profile->arr[i].sy = 0;	
+				
+				profile->arr[i].d_arr = coeff[0]+coeff[1]*profile->arr[i].zarr+coeff[2]*profile->arr[i].zarr*profile->arr[i].zarr;
+			}
+			break;
+
+		case 2: //ellipsoidal; side with largest radius has horizontal tangent, other side points towards focal_dist corresponding to smallest external radius
+			if(rad_ext[1] < rad_ext[0]){ //focussing alignment
+				slope = rad_ext[1] / focal_dist[1];
+				b = (-1.*(rad_ext[1]-rad_ext[0])*(rad_ext[1]-rad_ext[0])-slope*length*(rad_ext[1]-rad_ext[0])) / (slope*length+2.*(rad_ext[1]-rad_ext[0]));
+				k = rad_ext[0] - b;
+				a = sqrt((b*b*length)/(slope*(rad_ext[1]-k)));
+				for(i=0;i<profile->nmax;i++){
+					profile->arr[i].zarr = length/profile->nmax*i; //z coordinates, from 0 to length
+					profile->arr[i].profil = (rad_int[1]-rad_int[0])/length*profile->arr[i].zarr + rad_int[0]; //single capillary shape always conical
+					profile->arr[i].sx = 0; //set sx and sy to 0 as they are overwritten in start() anyway. 
+					profile->arr[i].sy = 0;	
+					profile->arr[i].d_arr = sqrt(b*b-(b*b*profile->arr[i].zarr*profile->arr[i].zarr)/(a*a))+k;
+				}
+			} else { //confocal (collimating) alignment
+				slope = rad_ext[0] / focal_dist[0];
+				b = (-1.*(rad_ext[0]-rad_ext[1])*(rad_ext[0]-rad_ext[1])-slope*length*(rad_ext[0]-rad_ext[1])) / (slope*length+2.*(rad_ext[0]-rad_ext[1]));
+				k = rad_ext[1] - b;
+				a = sqrt((b*b*length)/(slope*(rad_ext[0]-k)));
+				j = profile->nmax-1;
+				for(i=0;i<profile->nmax;i++){
+					profile->arr[i].zarr = length/profile->nmax*i; //z coordinates, from 0 to length
+					profile->arr[i].profil = (rad_int[1]-rad_int[0])/length*profile->arr[i].zarr + rad_int[0]; //single capillary shape always conical
+					profile->arr[i].sx = 0; //set sx and sy to 0 as they are overwritten in start() anyway. 
+					profile->arr[i].sy = 0;	
+					profile->arr[i].d_arr = sqrt(b*b-(b*b*profile->arr[j].zarr*profile->arr[j].zarr)/(a*a))+k;
```

Comment:
Dit gaat slecht aflopen.

Je zal dit moeten doen in twee for loops.

Reply:
Ah, ok. Nu snap ik het probleem :) Vond het niet direct
Good call. Fixed in a minute.

Reply:
👍 
Wat je ook had kunnen doen is ipv `j` te gebruiken te werken met `profile->nmax-1-i`